stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -831,7 +875,8 @@ class DiscardingTaskGroup: public TaskGroupBase {
 
 private:
   /// Resume waiting task with specified error
-  void resumeWaitingTaskWithError(SwiftError *error,
+  PreparedWaitingTask prepareWaitingTaskWithError(AsyncTask* waitingTask,
```

Comment:
The important change here is that we PREPARE, but don't guarantee that we'll run -- it depends what model we're running in, task-to-thread or not.

If the returned prepared task is not null, we'll schedule it, but only AFTER unlocking the group